### PR TITLE
(1765) Edit a regulatory authority

### PR DESCRIPTION
**Pull request by pezholio**
# Changes in this PR

This allows a regulatory authority to be edited. There's still some rough edges to work out once we get the drafts/approvals in place, but this is enough to get started 

## Screenshots of UI changes

image

image

image

image

image

### CODE REVIEW

File: cypress/integration/admin/organisations/edit.spec.ts [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
       cy.visit('/admin');
-    });
 
-    it('Allows an organisation to be edited', () => {
       cy.get('a').contains('Regulatory authorities').click();
 
       cy.readFile('./seeds/test/organisations.json').then((organisations) => {
         const organisation = organisations[0];
 
+        cy.wrap(organisation).as('organisation');
```
Comment (bryangaledxw) [inline cypress/integration/admin/organisations/edit.spec.ts:12]: Neat!

File: src/organisations/presenters/organisation.presenter.ts [hunk 2]
```diff
@@ -2,6 +2,13 @@ import { I18nService } from 'nestjs-i18n';
 import { Organisation } from './../organisation.entity';
 import { TableRow } from '../../common/interfaces/table-row';
 import { SummaryList } from '../../common/interfaces/summary-list';
+
+interface OrganisationSummaryListOptions {
+  classes?: string;
+  removeBlank?: boolean;
+  includeName?: boolean;
+  includeActions?: boolean;
+}
```
Comment (bryangaledxw) [inline src/organisations/presenters/organisation.presenter.ts:11]: Acknowledge this is picky, but would but a line of whitespace between the interface and class declarations

File: src/organisations/presenters/organisation.presenter.ts [hunk 3]
```diff
@@ -30,61 +37,100 @@ export class OrganisationPresenter {
...
-            ),
-          },
-          value: {
-            text: this.organisation.alternateName,
-          },
+  public async summaryList(
+    options: OrganisationSummaryListOptions = {},
+  ): Promise<SummaryList> {
+    const classes = options.classes || 'govuk-summary-list--no-border';
+    const removeBlank = options.removeBlank || false;
```
Comment (bryangaledxw) [inline src/organisations/presenters/organisation.presenter.ts:44]: Could also do `!!options.removeBlank` for succinctness, which we're doing elsewhere with the exception of entity constructors

File: src/organisations/presenters/organisation.presenter.ts [hunk 4]
```diff
@@ -30,61 +37,100 @@ export class OrganisationPresenter {
...
+
+    if (includeActions) {
+      rows = rows.map((row) => {
+        return {
+          ...row,
+          actions: {
+            items: [
+              {
+                href: `/admin/organisations/${this.organisation.slug}/edit`,
+                text: 'Change',
```
Comment (bryangaledxw) [inline src/organisations/presenters/organisation.presenter.ts:122]: Should this be translated?

File: src/common/interfaces/summary-list.ts [hunk 5]
```diff
@@ -1,6 +1,13 @@
+type SummaryListActionItem = {
+  href: string;
+  text: string;
+  visuallyHiddenText: string;
+};
+
 type SummaryListItem = {
   key: { text: string } | { html: string };
   value: { text: string } | { html: string };
+  actions?: { items: Array<SummaryListActionItem> };
```
Comment (bryangaledxw) [inline src/common/interfaces/summary-list.ts:10]: We're mixing `Foo[]` and `Array<Foo>` in the same file

File: src/organisations/admin/organisations.controller.ts [hunk 6]
```diff
@@ -87,9 +88,38 @@ export class OrganisationsController {
   async confirm(
     @Param('slug') slug: string,
     @Body() organisationDto: OrganisationDto,
-  ): Promise<Organisation> {
+  ): Promise<any> {
```
Comment (bryangaledxw) [inline src/organisations/admin/organisations.controller.ts:91]: Can we have an interface to encode what the `confirm.njk` template is expecting. Or even just `Promise<Organisation & {summaryList: SummaryList}>` here

File: src/organisations/admin/organisations.controller.ts [hunk 7]
```diff
@@ -58,4 +73,57 @@ export class OrganisationsController {
...
+        classes: 'govuk-summary-list',
+        removeBlank: false,
+        includeName: true,
+        includeActions: true,
+      }),
+    };
+  }
+
+  @Put('/:slug')
+  @Render('admin/organisations/confirmation')
```
Comment (bryangaledxw) [inline src/organisations/admin/organisations.controller.ts:119]: Is there more distinct naming we can use than `confirm` and `confirmation`?

**Review by bryangaledxw — APPROVED** [review]
Looks good. I've made a bunch of comments but all very minor. I think the one I care most about is not having the `Promise<any>` on `OrganisationsController`

File: cypress/integration/admin/organisations/edit.spec.ts [hunk 8]
```diff
@@ -0,0 +1,65 @@
+describe('Showing organisations', () => {
+  context('When I am logged in as admin', () => {
+    beforeEach(() => {
+      cy.loginAuth0('admin');
+      cy.visit('/admin');
+    });
+
+    it('Allows an organisation to be edited', () => {
+      cy.get('a').contains('Regulatory authorities').click();
```
[the remark this answers is not in the document]
Comment (pezholio) [reply cypress/integration/admin/organisations/edit.spec.ts:7]: I don't think it was translated at the time. I'll fix 👍

File: cypress/support/commands.ts [hunk 9]
```diff
@@ -83,3 +83,31 @@ Cypress.Commands.add(
...
+ * Check a form element's value
+ */
+
+Cypress.Commands.add('checkInputValue', (label: string, value: string) => {
+  return cy.translate(label).then((formLabel) => {
+    cy.get('body').should('contain', formLabel);
+    cy.get('label')
+      .contains(formLabel)
+      .siblings('input')
+      .then(($input) => {
```
[the remark this answers is not in the document]
Comment (pezholio) [reply cypress/support/commands.ts:97]: Prefixing with `$` shows that the thing we're using is an element. Looks to be more of a convention though, yeah

File: src/organisations/admin/organisations.controller.ts (hunk 6 above)
Comment (pezholio) [reply src/organisations/admin/organisations.controller.ts:91]: Whoops that was lazy of me. I meant to come back and fix it

**Pull request merged**